Cargo.toml

```diff
@@ -26,6 +22,7 @@ cortex-m = { version = "0.7.7", features = ["critical-section-single-core"] }
 defmt-rtt = "0.4.0"
 cortex-m-rt = "0.7.2"
 panic-probe = { version = "0.3.0", features = ["print-defmt"] }
+stm32g4xx-hal = { git = "https://github.com/usbalbin/stm32g4xx-hal", branch = "hrtim", features = ["defmt", "hrtim"] }
```

Comment:
@burrbull As far as I know we can not have optional dev dependendecies and thus need to select one hal to use. Would it then be better to let the examples live in the corresponding hals? See https://github.com/stm32-rs/stm32g4xx-hal/pull/192